### PR TITLE
The Backrooms mod

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
Mods "The Backrooms mod"

#### Purpose of change
Recently I [discovered](https://www.reddit.com/r/cataclysmdda/comments/p4m20y/i_made_an_infinite_backrooms_mod/) this awesome mod and decided to add it to list of in-repo mods as I feel it absolutely complies with our rules of adding mods to the repo.

Also I feel that this mod is a great example of what could be achieved by json alone, this might encourage other modders to create something great.

I read that at some time its author, @TGWeaver planned to add it to base game, but it seems that it was never accomplished. I contacted them and asked their permission to publish the mod, and I received this permission.

This is a total conversion mod which removes all normal mapgen and replaces it with endless rooms in accordance with the [lore](https://en.wikipedia.org/wiki/The_Backrooms). No cities, no roads, no forests, no rivers, no vehicles, no locations other than yellow rooms of various sizes and normal zombies.

#### Describe the solution
Majority of the work was done by @TGWeaver, I only fixed some errors, added `ERASE_ALL_BEFORE_PLACING_TERRAIN` flag to avoid new errors, removed some unused or unneeded stuff.

Also I made the `Backrooms` scenario the only one available to select (by whitelisting it). No other scenarios are supported. `Play Now! (Default Scenario)` also ain't supported. `Random Character` and ordinary `Play Now!` are supported.

#### Describe alternatives you've considered
None.

#### Testing
Created world with the mod, created a character, started the game without any errors.

#### Additional context
All the world will look like this:
изображение

изображение

This is the overmap
изображение

### CODE REVIEW

**Comment by Fris0uman** [comment]
doesn't that cause license issues because the Backroom is SCP content?

**Comment by Night-Pryanik** [comment]
AFAIK Backrooms is a concept invented by an anon on 4chan, so it can't licensed. Even if it was later integrated into SCP, SCP has CC-BY-SA-3.0. I'm not sure whether fair use can cause license issues.

**Comment by Fris0uman** [comment]
Wikipedia suggests you're right https://en.wikipedia.org/wiki/The_Backrooms and it's not a SCP thing

File: data/mods/Backrooms/items.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+[
+  {
+    "id": "jug_cooler",
+    "type": "GENERIC",
+    "category": "container",
+    "name": { "str": "water cooler jug" },
+    "looks_like": "bottle_plastic",
+    "description": "A sizable blue plastic jug meant to hold and supply water in a water cooler or similar dispenser.  Holds a lot of liquid, but unfortunately it doesn't come with a cap, making it hard to carry around when full.",
```
Comment (TheShadowFerret) [inline data/mods/Backrooms/items.json:8]: I see mainline content viability 👀

File: data/mods/Backrooms/mod_tiles.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+[
+  {
+    "type": "mod_tileset",
+    "compatibility": [ "ChestHole32", "MShock32Modded", "MshockXottoplus", "MSX++DEAD_PEOPLE", "UNDEAD_PEOPLE", "UNDEAD_PEOPLE_BASE" ],
```
Comment (Fris0uman) [inline data/mods/Backrooms/mod_tiles.json:4]: this and the png should not get in this PR. Maybe submit it to the tileset repo if you want it added but not directly to the game.

**Comment by anoobindisguise** [comment]
I gave this a few hours' playtime and it was fun, but the world needs to be moved to an underground layer because presently it is open to the sky and seemingly affected by ambient temperature and day/night cycle. If I try to climb I get a warning about either needing furniture or no terrain able to support my weight, rather than a ceiling message. I assume if portal storms happen it will be bad since there's hypothetically no cover. I could definitely see myself contributing to this in the future though, it's cool. It scratches that itch of wanting something like the lab escape scenario back when I first started playing.

**Comment by Night-Pryanik** [comment]
> the world needs to be moved to an underground layer because presently it is open to the sky and seemingly affected by ambient temperature and day/night cycle.

No need. The rooms all have roof above them which protect from weather.
изображение
изображение

> If I try to climb I get a warning about either needing furniture or no terrain able to support my weight, rather than a ceiling message.

Hmm, that's strange, I get the ceiling message:
изображение

**Comment by anoobindisguise** [comment]
Huh, you're right, whatever error I had seemed to go away after saving and loading. I could definitely build a stepladder and attempt to climb but it's not happening anymore. Any idea how it will treat seasons/temperature?

**Comment by Night-Pryanik** [comment]
> Any idea how it will treat seasons/temperature?

I think the seasons change and temperature rising/falling will go as they should. Players can set eternal seasons and eternal time of day to simulate completely stable and unchanging indoor environment.

File: data/mods/Backrooms/items.json (hunk 1 above)
Comment (bombasticSlacks) [reply data/mods/Backrooms/items.json:8]: I'm gonna resolve this Convo but I agree water coolers could be moved if there was desire.

**Pull request merged**